executor/containerdexecutor/executor_unix.go

```diff
@@ -0,0 +1,170 @@
+//go:build !windows
+// +build !windows
+
+package containerdexecutor
+
+import (
+	"context"
+	"os"
+	"runtime"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/mount"
+	containerdoci "github.com/containerd/containerd/oci"
+	"github.com/containerd/continuity/fs"
+	"github.com/docker/docker/pkg/idtools"
+	"github.com/moby/buildkit/executor"
+	"github.com/moby/buildkit/executor/oci"
+	"github.com/moby/buildkit/snapshot"
+	"github.com/moby/buildkit/util/bklog"
+	"github.com/moby/buildkit/util/network"
+	rootlessspecconv "github.com/moby/buildkit/util/rootless/specconv"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
+func getUserSpec(user, rootfsPath string) (specs.User, error) {
+	var err error
+	var uid, gid uint32
+	var sgids []uint32
+	if rootfsPath != "" {
+		uid, gid, sgids, err = oci.GetUser(rootfsPath, user)
+	} else {
+		uid, gid, err = oci.ParseUIDGID(user)
+	}
+	if err != nil {
+		return specs.User{}, errors.WithStack(err)
+	}
+	return specs.User{
+		UID:            uid,
+		GID:            gid,
+		AdditionalGids: sgids,
+	}, nil
+}
+
+func (w *containerdExecutor) prepareExecutionEnv(ctx context.Context, rootMount executor.Mount, mounts []executor.Mount, meta executor.Meta, details *jobDetails) (func(), string, string, error) {
```

Comment:
It isn't ideal that this is a mix of 2 things: preparing `/etc` files and setting up properties of `details` (otherwise I would call this `prepareEtcPaths`. If it is already a mix, then why is `CWD` different? If you don't have better ideas then rename the cwd to `prepareCWD` so it is consitent.